### PR TITLE
Add KODEBUG environment variable to trigger debug build

### PR DESCRIPTION
**Pull request by Frenzie**
@poire-z Any thoughts on the name?

### CODE REVIEW

File: Makefile.defs [hunk 1]
```diff
@@ -143,7 +143,9 @@ endif
 # Makefiles do some tricky stuff to differentiate HOST/TARGET
 BASE_CFLAGS:=-O2 -ffast-math -pipe -fomit-frame-pointer
 # Use this for debugging:
-#BASE_CFLAGS:=-O0 -g
+ifdef KODEBUG
+    BASE_CFLAGS:=-Og -g
```
Comment (Frenzie) [inline Makefile.defs]: Wait, is there a reason this line was different? Shouldn't it just be the same as regular (`-ffast-math` etc) but with `0g -g`?

Comment (poire-z) [reply Makefile.defs]: Dunno, may be updating it was just forgotten when updating the main one (check commit history maybe ?)
OK with `KODEBUG` , it sounds nice :)

**Comment by Frenzie** [comment]
@poire-z Any thoughts on the additions?

**Comment by poire-z** [comment]
Just the thought that I can't comment on the last commit, as I don't know much about luajit (and Makefile business). But OK with the other stuff.

**Pull request merged**